python/source.py

```diff
@@ -923,23 +964,22 @@ def incoming_mask(self, x0, y0, kx, ky, X):
 
         return incoming_points, outgoing_points, waist_points
 
+    def get_r_rhat(self, X, X0):
+        """Returns r and rhat before normalizing rhat to be used in green2d and get_fields for overflow prediction"""
```

Comment:
I created this new function to minimize the amount of repeated code because I now need the term $kr$ in `get_fields` as well as `green2d`.